dvc/repo/experiments/remove.py

```diff
@@ -12,41 +13,48 @@
 
 @locked
 @scm_context
-def remove(repo, exp_names=None, queue=False, **kwargs):
-    if not exp_names and not queue:
+def remove(repo, refs_or_revs=None, queue=False, **kwargs):
+    if not refs_or_revs and not queue:
         return 0
 
     removed = 0
     if queue:
         removed += len(repo.experiments.stash)
         repo.experiments.stash.clear()
-    if exp_names:
-        ref_infos = list(_get_exp_refs(repo, exp_names))
-        remove_exp_refs(repo.scm, ref_infos)
-        removed += len(ref_infos)
+    if refs_or_revs:
+        for ref_or_rev in refs_or_revs:
+            _remove_exp_by_ref_or_rev(repo, ref_or_rev)
+            removed += 1
     return removed
 
 
-def _get_exp_refs(repo, exp_names):
+def _get_exp_stash_index(repo, ref_or_rev: str) -> Optional[int]:
+    stash_ref_infos = repo.experiments.stash_revs
```

Comment:
`experiments.stash_revs` is not a list of `ref_infos`. It's a dictionary of stash entries (which are just loose individual git commits, not named git references): https://github.com/iterative/dvc/blob/38cd7d01c2fd14fc28c68f70bc963b105ce64db0/dvc/repo/experiments/__init__.py#L86